src/plugins/memoryvalue/memoryvalue.c

```diff
@@ -0,0 +1,188 @@
+/**
+ * @file
+ *
+ * @brief memoryvalue plugin
+ *
+ * @copyright BSD License (see LICENSE.md or https://www.libelektra.org)
+ *
+ */
+
+#define _XOPEN_SOURCE
+#include "memoryvalue.h"
+#include <kdberrors.h>
+#include <kdbhelper.h>
+#include <kdbtypes.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+static char * deblank (char * input)
+{
+	kdb_unsigned_long_long_t i, j;
+	char * output = input;
+	for (i = 0, j = 0; i < strlen (input); i++, j++)
+	{
+		if (input[i] != ' ')
+			output[j] = input[i];
+		else
+			j--;
+	}
+	output[j] = 0;
+	return output;
+}
+
+static kdb_unsigned_long_long_t isValidKey (Key * key)
+{
+
+	const char * value = keyString (key);
+	// else we manipulate the original
+	char * tempval = elektraStrDup (value);
+
+	kdb_unsigned_long_long_t ret;
+
+	char * endPtr;
+	// convert to long, if valid key, pointer should point to spaces or memory suffix like MB, GB
+	ret = ELEKTRA_UNSIGNED_LONG_LONG_S (tempval, &endPtr, 10);
```

Comment:
Check if this was successful? (Any number found?) Please also add a testcase with simply "B".

Reply:
The problem with ELEKTRA_UNSIGNED_LONG_LONG_S, as I have understood it is, that the behavior is not defined in case of an error.

Reply:
`ELEKTRA_UNSIGNED_LONG_LONG_S` is just `strtoull(3)` or `strtoul(3)`, IIRC. Their behaviour in case of error is quite well defined.

Reply:
I know, but I thought it is not if the string specified is not valid.
http://www.cplusplus.com/reference/cstdlib/strtoull/

Reply:
Or should I just expect the string to be valid but simply not contain a number?
Then I can verfiy the conversion with 0ULL